drivers/vmwarevsphere/flags.go

```diff
@@ -239,8 +239,8 @@ func (d *Driver) UnmarshalJSON(data []byte) error {
 		d.IP = driverOpts.String("vmwarevsphere-vcenter")
 	}
 
-	if _, ok := driverOpts.Values["vmwarevsphere-user"]; ok {
-		d.Username = driverOpts.String("vmwarevsphere-user")
+	if _, ok := driverOpts.Values["vmwarevsphere-username"]; ok {
+		d.Username = driverOpts.String("vmwarevsphere-username")
```

Comment:
Fixing a lookup here using a non-existent key. The effect of this bug is that the username simply doesn't get updated if the user tries to change it for an existing machine (i.e. the same thing Rancher machine always did in the past).